index.rst

```diff
@@ -1,61 +1,263 @@
-..
-  Technote content.
+:tocdepth: 1
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+Abstract
+========
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+This technote gathers information about the history, current state and contemplates future evolutions of the Vera Rubin Observatory Control System (Rubin-OCS) Middleware.
+The middleware is the backbone of the Rubin-OCS.
+The highly distributed nature of the Rubin-OCS places tight constraints in terms of latency, availability and reliability for the middleware.
+Here we gather information to answer some common questions regarding technology choices, describe some of the in-house work done to obtain a stable system and document some of the concerns with the current state of the system, potential impacts for near-future/commissioning and future/operations.
+We also cover some of the work we have been doing to investigate alternative technologies and suggest some potential roadmaps for the future of the system.
 
-  Use the following syntax for sections:
+Introduction
+============
 
-  Sections
-  ========
+"Middleware" is the term used to describe software applications used for communication in sofware systems.
+These technologies became popular with the advent of *distributed systems*, wich come as a solution to the problem of parallel computation.
+As software and hardware systems became more and more complex, it become impractical to develop and execute them in a single process and, eventually, a single node.
+With that, software evolved from *monolith* applications, where a single program executes in a single process or node, to *distributed* applications, where the system is divided into a number of smaller applications; each running on their own process or node.
+For these applications to work together coherently they must be able to comunicate with each other, thus giving origin to middleware technologies.
 
-  and
+How a *distributed system* is broken down into smaller pieces is heavily dependent uppon the problem.
+Some systems are only broken down into a small number of components each still in charge of large contexts, others are broken down into several small applications that are in charge only of small simple tasks.
```

Comment:
I suggest "broken down into many small applications" to make a clearer contrast.